Common/include/containers/CLookUpTable.hpp

```diff
@@ -0,0 +1,160 @@
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2022, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <iomanip>
+#include <string>
+#include <vector>
+
+#include "../../Common/include/option_structure.hpp"
+#include "../../Common/include/linear_algebra/blas_structure.hpp"
+#include "../../Common/include/toolboxes/CSquareMatrixCM.hpp"
+#include "CFileReaderLUT.hpp"
+#include "CTrapezoidalMap.hpp"
+
+class CLookUpTable {
+ protected:
+  int rank; /*!< \brief MPI Rank. */
+
+  std::string file_name_lut;
+  std::string type_lut;
+  std::string version_lut;
+  std::string version_reader;
+  unsigned long n_points;
+  unsigned long n_triangles;
+  unsigned long n_variables;
+  unsigned long n_hull_points;
+
+  /*! 
+   * \brief the lower and upper limits of the enthalpy and progress variable 
+   */
+  su2double limits_table_enth[2];
+  su2double limits_table_prog[2];
+
+  /*! \brief Holds the variable names stored in the table file.
+   * Order is in sync with data
+   */
+  std::vector<std::string> names_var;
+
+  /*! \brief
+   * Holds all data stored in the table. First index addresses the variable
+   * while second index addresses the point.
+   */
+  su2activematrix table_data;
+
+  su2matrix<unsigned long> triangles;
+
+  /* we do not know this size in advance until we go through the entire lookup table */
+  std::vector<std::vector<unsigned long> > edges;
+  std::vector<std::vector<unsigned long> > edge_to_triangle;
+
+  /*! \brief 
+   * The hull contains the boundary of the lookup table 
+   */
+  std::vector<unsigned long> hull;
+
+  CTrapezoidalMap trap_map_prog_enth;
+
+  std::vector<su2activematrix> interp_mat_inv_prog_enth;
+
+  inline int GetIndexOfVar(const std::string& nameVar) {
+    int index = find(names_var.begin(), names_var.end(), nameVar) - names_var.begin();
+    if (index == int(names_var.size())) {
+      index = -1;
+      std::string error_msg = "Variable '";
+      error_msg.append(nameVar);
+      error_msg.append("' is not in the lookup table.");
+      SU2_MPI::Error(error_msg, CURRENT_FUNCTION);
+    }
+    return index;
+  }
+
+/* get the pointer to the column data of the table (density, temperature, source terms, ...) */
+inline const su2double* GetDataP(const std::string& name_var) {
```

Comment:
```suggestion
  /*!
   * \brief Get the pointer to the column data of the table (density, temperature, source terms, ...)
   */
  inline const su2double* GetDataP(const std::string& name_var) const {
```

Reply:
OK, I get it, I guess the basic guideline should be:
- make everything const except when you can't
- tell class functions that they do not change member variables unless they do.